pytorch_lightning/metrics/metric.py

```diff
@@ -142,7 +142,8 @@ def forward(self, *args, **kwargs):
         Automatically calls ``update()``. Returns the metric value over inputs if ``compute_on_step`` is True.
         """
         # add current step
-        self.update(*args, **kwargs)
+        with torch.no_grad():
```

Comment:
@teddykoker @SkafteNicki  also when we are using dist_sync_on_step as True, that should be a no_grad op. Since DDP will expect grads to be computed only for this GPU and then sync those when backpropagating.

So https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/utilities/distributed.py#L76 needs to be done in no_grad if I am not wrong.

Reply:
If we do that will it not break backpropagation, as everything gets detached from the computational graph?
We should probably be able to test this.

Reply:
I don't think this should be `no_grad` for the exact same reason @SkafteNicki mentioned. If you backdrop, the device-changes (as well as dip sync) need to be recorded by the graph to make sure you can go back to the original device during backdrop (and to also find all the activations there).